src/Compilers/CSharp/Portable/Binder/Semantics/OverloadResolution/MethodTypeInference.cs

```diff
@@ -2878,22 +2881,9 @@ static bool isExpressionType(TypeSymbol? type)
                 return false;
             }
 
-            static HashSet<TypeWithAnnotations>? removeTypes(HashSet<TypeWithAnnotations>? types, Func<TypeWithAnnotations, bool> predicate)
+            static void removeTypes(HashSet<TypeWithAnnotations>? types, Func<TypeWithAnnotations, bool> predicate)
```

Comment:
>Func<TypeWithAnnotations, bool> [](http://example.com/codeflow?start=73&length=31)

Consider changing this to `Predicate<TypeWithAnnotations>` to avoid the extra delegate below.

Reply:
This change moved to #69070, and suggestion is applied there.